test/integration_test.exs

```diff
@@ -0,0 +1,251 @@
+              }
+            }
+          ]
+        })
+
+      assert conn.state == :sent
+      assert conn.status == 200
+      assert get_resp_header(conn, "content-type") == ["application/json; charset=utf-8"]
+      assert conn.resp_body == expected_body
+    end
+  end
+
+  describe "when fetching a component returns a 404" do
+    test "it returns a 200, with the failing component status set as 404 and its original id still set" do
+      json_body = ~s({
+        "components": [{
+                        "id": "news-front-page",
+                        "endpoint": "localhost:8082/non_200_status/404",
+                        "must_succeed": true
+                        }]
+                      }
+                    )
+      conn = conn(:post, "/collect", json_body)
+      conn = Router.call(conn, @opts)
+
+      expected_body =
+        Jason.encode!(%{
+          components: [
+            %{
+              status: 404,
+              index: 0,
+              id: "news-front-page",
+              envelope: %{
+                head: [],
+                bodyLast: [],
+                bodyInline: ""
+              }
+            }
+          ]
+        })
+
+      assert conn.state == :sent
+      assert conn.status == 200
+      assert get_resp_header(conn, "content-type") == ["application/json; charset=utf-8"]
+      assert conn.resp_body == expected_body
+    end
+  end
+
+  describe "when fetching the component times out" do
+    test "it returns a 200, with the failing component status set as 408 and its original id still set" do
+      json_body = ~s({
+        "components": [{
+                        "id": "weather-front-page",
+                        "endpoint": "localhost:8082/non_200_status/408",
+                        "must_succeed": true
+                        }]
+                      }
+                    )
+      conn = conn(:post, "/collect", json_body)
+      conn = Router.call(conn, @opts)
+
+      expected_body =
+        Jason.encode!(%{
+          components: [
+            %{
+              status: 408,
+              index: 0,
+              id: "weather-front-page",
+              envelope: %{
+                head: [],
+                bodyLast: [],
+                bodyInline: ""
+              }
+            }
+          ]
+        })
+
+      assert conn.state == :sent
+      assert conn.status == 200
+      assert get_resp_header(conn, "content-type") == ["application/json; charset=utf-8"]
+      assert conn.resp_body == expected_body
+    end
+  end
+
+  describe "when fetching the component receives a server error" do
+    test "it returns a 200, with the failing component status set as 500 and its original id still set" do
+      json_body = ~s({
+        "components": [{
+                        "id": "weather-front-page",
+                        "endpoint": "localhost:8082/non_existent_path",
```

Comment:
this sounds like mocking a 404 to me

Reply:
you should use: `get "/non_200_status/:code" do`

Reply:
sure